lambda/write-activity-log/write-activity-log.ts

```diff
@@ -0,0 +1,80 @@
+import { SQSEvent } from "aws-lambda";
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import {
+  DynamoDBDocumentClient,
+  PutCommand,
+  PutCommandOutput,
+} from "@aws-sdk/lib-dynamodb";
+import {
+  SendMessageCommand,
+  SendMessageRequest,
+  SQSClient,
+} from "@aws-sdk/client-sqs";
+import { Activity, ActivityLogEntry } from "./models";
+
+const dynamoClient = new DynamoDBClient({});
+const dynamoDocClient = DynamoDBDocumentClient.from(
+  dynamoClient
+);
+
+const sqsClient = new SQSClient({});
+
+export const validateActivityLogEntry = (activityLogEntry: ActivityLogEntry): void => {
+  if (!(
+          activityLogEntry.user_id  !== undefined &&
+          activityLogEntry.session_id !== undefined &&
+          activityLogEntry.timestamp !== undefined &&
+          activityLogEntry.truncated !== undefined
+      )) {
+      throw new Error(`Could not validate activity log entry ${JSON.stringify(activityLogEntry)}`);
+  }
+};
+
+export const validateActivities = (activities: Activity[]): void => {
+    for (let i = 0; i < activities.length; i += 1) {
```

Comment:
Could we use a `for-of` loop here 
Sonar usually complains about these as code smells 
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&id=di-account-management-backend-write-user-services&open=AYYwXbK98mzb7uqxZ1VG&tab=why

Reply:
good suggestion